bus/src/main/java/org/hawkular/metrics/component/publish/InsertedDataSubscriber.java

```diff
@@ -131,5 +149,43 @@ void shutdown() {
         if (subscription != null) {
             subscription.unsubscribe();
         }
+        publishTask.cancel();
+        publishTimer.cancel();
+    }
+
+    private int getPublishPeriod() {
+        try {
+            return Integer.parseInt(publishPeriodProperty);
+        } catch (NumberFormatException e) {
+            log.warnf("Invalid publish period. Setting default value %s", METRICS_PUBLISH_PERIOD.defaultValue());
+            return Integer.parseInt(METRICS_PUBLISH_PERIOD.defaultValue());
+        }
+    }
+
+    private class PublishTask extends TimerTask {
```

Comment:
If there is an alternative to do the same here, I am fine with that.
Perhaps I'm re-implementing some pattern that is well known in the RxJava.
As this PR is relatively small, could you send a commit with that proposal on it ?